### PR TITLE
Added the spf13/cobra library to make the program a CLI program

### PR DESCRIPTION
**Pull request by renormalize**
* Added the `NewObsidianParserCommand()` function which creates the parser command which is `Execute()`-ed to run the program in `main()`.

* Changed package structure to adhere to Go conventions.

The package structure changes after adding the CLI library and thus the merge won't be automatic.

### CODE REVIEW

File: pkg/cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+			// TODO:could be improved to decide code flow based on state of the program instead of file path
+			if len(outputFilePath) > 0 {
+				obsidian.WriteToObsidianVault(formattedExpenses, outputFilePath)
+			} else {
+				obsidian.WriteToConsole(formattedExpenses)
+			}
+		},
+	}
+	// TODO @kartikx is the input file is always required from the user?
```
Comment (kartikx) [inline pkg/cmd/root.go:39]: Input file contains expenses, I am fine with keeping it always required for now.

File: main.go [hunk 2]
```diff
@@ -1,28 +1,11 @@
...
-		return
-	}
-
-	if len(OutputFilePath) > 0 {
-		writeToObsidianVault(formattedExpenses, OutputFilePath)
-	} else {
-		writeToConsole(formattedExpenses)
-	}
+	// run obsidian-finances-parser
+	obsidianParserCommand := cmd.NewObsidianParserCommand()
```
Comment (kartikx) [inline main.go:9]: How do we pass in the -input and -output flags here? As users, we can do:
```
ofp -input="input" -output="output"
```

But how do we do this in development?

File: pkg/cmd/root.go (hunk 1 above)
Comment (renormalize) [reply pkg/cmd/root.go:39]: Cool

File: main.go (hunk 2 above)
Comment (renormalize) [reply main.go:9]: Originally the filepaths for input and output were hardcoded in `FinanceStatementFilePath` and `OutputFilePath`. Now, not passing the flags to the binary simply makes use of these hardcoded values when the binary is run - as it did previously.

These path strings are passed in as default values when the flags are created [here](https://github.com/renormalize/obsidian-finances-parser/blob/8a849afe02ae424ed2daa8209b8f7125365ba1ba/pkg/cmd/root.go#L41-L42).

Does this answer your question?

**Comment by renormalize** [comment]
@kartikx rebased onto master, PTAL.

File: main.go (hunk 2 above)
Comment (kartikx) [reply main.go:9]: Sounds good.

**Pull request merged**